radix_sort.py

```diff
@@ -0,0 +1,78 @@
+from Queue import Queue
+import random
+import timeit
+
+
+def radix_sort(a_list):
+    queue_list = [Queue() for i in range(10)]
+    maximum = 0
+    for item in a_list:
+        if item > maximum:
+            maximum = item
+        queue_list[item % 10].put(item)
+    a_list_pointer = 0
+    for each_queue in queue_list:
+        while not each_queue.empty():
+            a_list[a_list_pointer] = each_queue.get()
+            a_list_pointer += 1
+
+    max_digits = len(str(maximum))
+    _radix_sort(a_list, max_digits, queue_list)
+
+
+def _radix_sort(a_list, max_digits, queue_list):
+    for i in xrange(1, max_digits):
+        for item in a_list:
+            queue_list[(item // 10**i) % 10].put(item)
+        a_list_pointer = 0
+        for each_queue in queue_list:
+            while not each_queue.empty():
+                a_list[a_list_pointer] = each_queue.get()
+                a_list_pointer += 1
+
+
+def string_radix_sort(string_list):
+    maximum = 0
+    for string in string_list:
+        if len(string) > maximum:
+            maximum = len(string)
+    queue_list = [Queue() for i in range(27)]
+
+    _string_radix_sort(string_list, maximum, queue_list)
+
+
+def _string_radix_sort(string_list, maximum, queue_list):
+    for i in reversed(xrange(0, maximum)):
```

Comment:
reversed is going to go ahead and materialize the `xrange` as a list, so you sort of lose the benefit of xrange here.  Might consider something like this instead:

```python
import itertools

for i in count(maximum - 1, -1):
    if i < 0:
        break
    ....
```